### ISSUE TITLE
DjVU bookmarks and thumbnails are not shown properly

### ISSUE THREAD

**Issue by sambul13** [sumatrapdfreader/sumatrapdf#1311]
When opening a DjVU book in SumatraPDF created with DjVU Document Express or similar package, its bookmarks and thumbnails are not displayed properly, thus making the book near impossible to navigate or work with. See examples with screenshots [here](https://forum.ru-board.com/topic.cgi?forum=5&topic=22740&start=2194&limit=1&m=1#1).

**Comment by GitHubRulesOK** [comment]
The bookmarks and thumbnails are within Document Express it does not export them in the DjVu ! so SumatraPDF can only show what is given in a Standard DjVu file 
![image](https://user-images.githubusercontent.com/12754913/70634599-6b24cf00-1c2a-11ea-90cc-041e080951f5.png)

**Comment by sambul13** [comment]
Document Express allows to Integrate Thumbnails. Those are still not shown by Sumatra. As to Bookmarks, some utilities allow to add those to DjVU document, like [DjVuTitleMaker](https://yadi.sk/d/uldYlyTbzzoxe). Those changes to the book numbers are treated as Bookmarks by Document Express if you look on these [screenshots](https://forum.ru-board.com/topic.cgi?forum=5&topic=22740&start=2194&limit=1&m=1#1).

But Sumatra shows them in small "Page number" field, impossible to read instead of showing in the vertical sidebar as Document express does (see the screenshots). If you mean adding feature to Sumatra to add bookmarks, for compatibility reasons it better match the way DjVUTitleMaker adds the bookmarks. 

Another important note, Sumatra should allow to view the bookmarks in the sidebar in Expended mode by a keyboard shortcut, when all multilevel bookmarks are expended for easy orientation within document like folder tree in File Explorer.

**Comment by GitHubRulesOK** [comment]
@kjk as you know djvu is progressive there are no "thumbnails" in the file they have to be built by the viewer. However SumatraPDF is not Showing the "Document Outline" correctly here is the DjView.
![image](https://user-images.githubusercontent.com/12754913/70647586-2eb09d80-1c41-11ea-8350-fd8d032a0211.png)

**Comment by sambul13** [comment]
As I mentioned above, DjVuTitleMaker integrates bookmarks (changed page numbers with added comments in this case) within DjVu document itself. Due to that fact, you were able to show them on your above picture, since nothing was shared except one DjVu book file, no XMLs or Registry exports.

As well, did you look at this [example](http://www.filedropper.com/djvubook1) I linked above? This file has Thumbnails integrated into it, they are not generated by the Viewer at file open time. This feature Integrate Thumbnails is provided by Doc Express.

So Sumatra should allow to view and create both integrated thumbnails like Doc Express and integrated bookmarks like DjVuTitleMaker, so it can be easily transferable and compatible with other viewers like [WinDjView](https://sourceforge.net/projects/windjviewextended/files/) that shows both by just opening the DjVu file without any XML or registry records.

**Comment by GitHubRulesOK** [comment]
A Document Maker / Editor can alter the "Outline" (TOC) of any document
Just as commercial Acrobat and Word do to .pdf and .docx
HOWEVER **a Reader, so in this case SumatraPDF Reader** or DjView or DjLibre can each keep their own local "Bookmarks" either in a settings file like an ini or the registry they should NOT alter the file.

In SumatraPDF it is confused by the "Outline" should be shown in the "Bookmarks" field and the Bookmarks can then be stored localy as "Favorites" in the settings file.

**Comment by sambul13** [comment]
I agree that Sumatra as a **Reader** shown mostly concentrate on showing content already integrated into the document, including both integrated thumbnails and bookmarks. Absolutely you're right.

**Comment by GitHubRulesOK** [comment]
I am  corrected, 😊  On checking the spec, just like any other IFF there are FORM:THUM fields for thumbnails though when working with TIFF I never found them of much value. It is better to respect the optional NAVM field _"If present, must immediately follow the DIRM chunk. Contains a BZZ-compressed outline of the document. "_

**Comment by kjk** [comment]
Test files no longer availble
